neuraxle/hyperparams/space.py

```diff
@@ -226,6 +228,76 @@ class HyperparameterSamples(RecursiveDict):
+        return CompressedHyperparameterSamples(self)
+
+
+@dataclass
+class CompressedHyperparameter:
+    """CompressedHyperParameter"""
+    step_name: str
+    hyperparams: dict
+    ancestor_steps: list
+
+
+class CompressedHyperparameterSamples:
+    """
+    Short-hand representation of `HyperparameterSamples`
+    """
+
+    def __init__(self, hps: HyperparameterSamples):
+        """Takes in `HyperparameterSamples` object and generates a compressed _seq."""
+        if not hps or not isinstance(hps, HyperparameterSamples):
+            raise ValueError("pass a valid `HyperparameterSamples` object")
+        self._flat_hps: dict = hps.to_flat_dict()  # this will be used to reconstruct the hyperparameters from
+        # compressed format
+        self._separator: str = hps.separator
+        self._seq: List[dict] = self._convert()
+
+    def __str__(self) -> str:
+        """Prints the `CompressedHyperparameterSamples._seq`."""
+        return f"{self._seq}"
+
+    def _group_hps_by_step(self, flat_steps_hps: ItemsView) -> 'OrderedDict[str, dict]':
+        """
+        Groups hyperparams
+        :param flat_steps_hps: Flat list of steps and hyper parameter
+        :return: OrderedDict containning grouped hyperparams as values and keys as pipeline stages & steps
+
+        >>>hyper_params = [('AddFeatures__PCA__copy', True), ('AddFeatures__PCA__iterated_power', 'auto')]
+        >>>self._group_hps_by_step(hyper_params = hyper_params)
+        ...OrderedDict([('AddFeatures__PCA', dict([('copy', True), ('iterated_power', 'auto')]))])
+        """
+        grouped_hyper_params = OrderedDict()
+        for steps_and_hps, hyper_param_val in flat_steps_hps:
+            *steps, hyper_param_key = steps_and_hps.split(self._separator)
+            grouped_hyper_params.setdefault(f"{self._separator}".join(steps), dict()).update({
+                hyper_param_key: hyper_param_val})
+
+        return grouped_hyper_params
+
+    def _convert_to_compressed_format(self, group_by_step: 'OrderedDict[str, dict]') -> 'List[dict]':
+        """Converts grouped hyper params to Compressed format."""
+        compressed = []
+        for steps in group_by_step:
+            *prev_steps, current_step = steps.split(self._separator)
+            hps: dict = group_by_step[steps]
+
+            compressed.append(CompressedHyperparameter(step_name=current_step, hyperparams=hps,
+                                                       ancestor_steps=prev_steps).__dict__)
+        return compressed
+
+    def _convert(self) -> 'List[dict]':
+        grouped_result: OrderedDict[str, dict] = self._group_hps_by_step(flat_steps_hps=self._flat_hps.items())
+        return self._convert_to_compressed_format(group_by_step=grouped_result)
+
+    def remove_parents(self) -> 'CompressedHyperparameterSamples':
+        raise NotImplementedError()
+
+    def wildcards(self) -> OrderedDict:
+        raise NotImplementedError()
```

Comment:
It would be interesting to see them added in the current pull request. If it is not added in the current pull request, then it is better to not add them at all.

Reply:
I will be adding them, in the current PR itself. As we agreed to have it in our discussions. I am still in the process to figure out the logic for the **wildcards** method. Most probably by today (or) tomorrow i should come to some conclusion.

Reply:
I've just found this. 
https://stackoverflow.com/questions/60915980/shortest-unique-presentable-filepath-for-each-path-in-a-list-of-paths
I didn't read the answer as they seem overly technical and precise, but this page could generate ideas on how to solve this.